docs/end_to_end_tutorials/dev_practices/e2e_evaluation.md

```diff
@@ -0,0 +1,41 @@
+# End-to-End Evaluation
+## Setting up an Evaluation Set
```

Comment:
I think we should better motivate what an "e2e dataset" looks like and how that's different than an isoalted "retrieval" dataset

Reply:
I think it's best that we create examples of doing this first.

Reply:
Added issue: https://github.com/jerryjliu/llama_index/issues/7310

Reply:
yeah we can mark stuff as a WIP, but still talk about the high-level concepts? 

like user is going to ask: why do i need automated dataset generation tools for e2e evaluation but use existing retrieval benchmarks for component-wise eval

i'm not saying we have all the answers right now. but we should at least provide some context for them to understand our thought process

Reply:
> use existing retrieval benchmarks for component-wise eval

I don't think this is going to be the recommended - it's possible to provide autogen for retrieval as well.

Reply:
Should we mention autogen for retrieval? (or should we add as a PR actually, that would be a good idea)

Reply:
Heads up I'm thinking of creating a `Dataset` abstraction that formalizes the common patterns (retrieval/relevance, QA) and provides some helper functions. We can link that once it's ready too.